test/suites/consolidation/suite_test.go

```diff
@@ -65,7 +66,119 @@ var _ = BeforeEach(func() {
+			nodePool.Spec.Disruption.Budgets = []karpv1.Budget{
+				{
+					Nodes: "0%",
+				},
+			}
+			env.ExpectCreated(nodeClass, nodePool, dep)
+
+			nodeClaims := env.EventuallyExpectCreatedNodeClaimCount("==", 1)
+			env.EventuallyExpectCreatedNodeCount("==", 1)
+			env.EventuallyExpectHealthyPodCount(selector, int(numPods))
+
+			nodeClaim := env.ExpectExists(nodeClaims[0]).(*karpv1.NodeClaim)
+			lastPodEventTime := nodeClaim.Status.LastPodEventTime
+
+			// wait 10 seconds so that we don't run into the de-dupe timeout
+			time.Sleep(10 * time.Second)
+
+			dep.Spec.Replicas = lo.ToPtr[int32](4)
+			By("removing one pod from the node")
+			env.ExpectUpdated(dep)
+
+			Eventually(func(g Gomega) {
+				nodeClaim = env.ExpectExists(nodeClaim).(*karpv1.NodeClaim)
+				g.Expect(nodeClaim.Status.LastPodEventTime.Time).ToNot(BeEquivalentTo(lastPodEventTime.Time))
+			}).WithTimeout(5 * time.Second).WithPolling(1 * time.Second).Should(Succeed())
+			lastPodEventTime = nodeClaim.Status.LastPodEventTime
+
+			// wait 10 seconds so that we don't run into the de-dupe timeout
+			time.Sleep(10 * time.Second)
+
+			dep.Spec.Replicas = lo.ToPtr[int32](5)
+			By("adding one pod to the node")
+			env.ExpectUpdated(dep)
+
+			Eventually(func(g Gomega) {
+				nodeClaim = env.ExpectExists(nodeClaim).(*karpv1.NodeClaim)
+				g.Expect(nodeClaim.Status.LastPodEventTime.Time).ToNot(BeEquivalentTo(lastPodEventTime.Time))
+			}).WithTimeout(5 * time.Second).WithPolling(1 * time.Second).Should(Succeed())
+		})
+		It("should update lastPodEventTime when pods go terminal", func() {
+			podLabels := map[string]string{"app": "regular-app"}
+			pod := test.Pod(test.PodOptions{
+				// use a non-pause image so that we can have a sleep
+				Image:        "alpine:3.20.2",
+				Command:      []string{"/bin/sh", "-c", "sleep 30"},
+				PreStopSleep: lo.ToPtr(int64(30)),
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: podLabels,
+				},
+				ResourceRequirements: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+				},
+				RestartPolicy: corev1.RestartPolicyNever,
+			})
+			job := &batchv1.Job{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      test.RandomName(),
+					Namespace: "default",
+				},
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: pod.ObjectMeta,
+						Spec:       pod.Spec,
+					},
+				},
+			}
+			selector := labels.SelectorFromSet(podLabels)
+			nodePool.Spec.Disruption.Budgets = []karpv1.Budget{
+				{
+					Nodes: "0%",
+				},
+			}
+			env.ExpectCreated(nodeClass, nodePool, job)
+
+			nodeClaims := env.EventuallyExpectCreatedNodeClaimCount("==", 1)
+			env.EventuallyExpectCreatedNodeCount("==", 1)
+			pods := env.EventuallyExpectHealthyPodCount(selector, int(1))
+
+			nodeClaim := env.ExpectExists(nodeClaims[0]).(*karpv1.NodeClaim)
+			lastPodEventTime := nodeClaim.Status.LastPodEventTime
+
+			Eventually(func(g Gomega) {
```

Comment:
What's the mechanism that we are using here to make sure that we have enough time between the pod creation and the pod completion?

Reply:
I'm not sure I understand. The pod itself is sleeping for 30 seconds, and the timeout is one minute. Isn't that enough? We also only start the timer after its gone healthy, which means it's started running.

Reply:
I was mainly just trying to grok that and confirm since it's not immediately clear to me from reading through it at a high-level. Makes sense to me. Some in-code documentation might help here but not blocking